src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -0,0 +1,208 @@
+
+            IList<PackageSource> listEndpoints = GetEndpointsAsync();
+
+            foreach (PackageSource source in listEndpoints)
+            {
+                string target = source.Source;
+                string name = source.Name;
+
+                SourceRepository repository = Repository.Factory.GetCoreV3(target);
+                PackageSearchResource resource = await repository.GetResourceAsync<PackageSearchResource>();
+
+                if (resource is null)
+                {
+                    continue;
+                }
+
+                IEnumerable<IPackageSearchMetadata> results = await resource.SearchAsync(
+                    string.Join("+", Arguments).Trim().Replace(' ', '+'), // The arguments (query strings) are joined with '+' so that the queries are passed to the
+                                                                          // search query url with the correct format (eg. "logging extensions" --> "logging+extensions")
+                    searchFilter,
+                    skip: 0,
+                    take: Take,
+                    logger,
+                    cancellationToken);
+
+                Console.WriteLine(new string('=', 20));
+                Console.WriteLine($"Source: {source.Name}");
+
+                if (!results.Any())
+                {
+                    Console.WriteLine(new string('-', 20));
+                    Console.WriteLine("No results found.");
+                    Console.WriteLine(new string('-', 20) + "\n");
+                }
+                else
+                {
+                    if (Verbosity == Verbosity.Quiet)
+                    {
+                        System.Console.WriteLine($"Source: {source.Name}");
+                        System.Console.WriteLine(new string('-', 20));
+                    }
+                    PrintResults(results);
+                }
+            }
+        }
+
+        private void PrintResults(IEnumerable<IPackageSearchMetadata> results)
+        {
+            foreach (IPackageSearchMetadata result in results)
+            {
+                Console.WriteLine(new string('-', 20));
+
+                CultureInfo culture = CultureInfo.CurrentCulture;
+
+                string printBasicInfo = $"> {result.Identity.Id} | {result.Identity.Version.ToNormalizedString()}";
+
+                string downloads, printDownloads;
+
+                if (result.DownloadCount != null)
+                {
+                    downloads = string.Format(culture, "{0:N}", result.DownloadCount);
+                    printDownloads = $" | Downloads: {downloads.Substring(0, downloads.Length - 3)}";
+                }
+                else
+                {
+                    printDownloads = " | Downloads: N/A";
+                }
+
+                System.Console.WriteLine(Verbosity != Verbosity.Quiet ? printBasicInfo + printDownloads : printBasicInfo); // System.Console is used so that output is not suppressed by Verbosity.Quiet
```

Comment:
I like that Quiet only shows the basic info! Should anything else only show basic? Like Minimal? I'd imagine Normal and above would show descriptions/etc.

Reply:
This `Verbosity` property only has three types: Quiet, Normal, and Detailed ([see here](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.CommandLine/Common/Verbosity.cs)). The Verbosity you're referring to might be the MSBuild verbosity levels (I might be wrong about this), and I'm not sure how to, and whether I even should, adapt the output to those Verbosity levels.